src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/AspNetCoreEnvironmentTelemetryInitializer.cs

```diff
@@ -8,7 +8,7 @@
     /// <see cref="ITelemetryInitializer"/> implementation that stamps ASP.NET Core environment name
     /// on telemetries.
     /// </summary>
-    internal class AspNetCoreEnvironmentTelemetryInitializer: ITelemetryInitializer
+    public class AspNetCoreEnvironmentTelemetryInitializer: ITelemetryInitializer
```

Comment:
(This comment applies to all the newly public classes.)
Should these types be `sealed`? It's usually a good idea with public types that were not designed to be extended by users - especially if they have any virtual methods.

Reply:
I agree it can be made sealed. However, similar ones in Web repo are not sealed.

https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/Web/Web.Shared.Net/SessionTelemetryInitializer.cs#L12

I'll limit my change to just making classes public for now. When we address (https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/612) this can be revisited.